### PR TITLE
[3.12.1] Fix some transaction::Methods lifetime issues

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

There are some lifetime issues in the transaction methods, which in some situations can lead to UB during certain document operations on DBServers.

- [X] :hankey: Bugfix

### Checklist

- [x] :book: CHANGELOG entry made

### CODE REVIEW

File: arangod/Transaction/Methods.cpp [hunk 1]
```diff
@@ -4008,8 +4003,7 @@ Future<OperationResult> Methods::removeInternal(
   if (_state->isCoordinator()) {
     f = removeCoordinator(collectionName, value, options, api);
   } else {
-    OperationOptions optionsCopy = options;
-    f = removeLocal(collectionName, value, optionsCopy);
+    f = removeLocal(collectionName, value, options);
```
Comment (jsteemann) [inline arangod/Transaction/Methods.cpp:4006]: nit: is it possible to std::move the `options` here into `removeLocal(...)`? That would save a copy of the options object, which contains strings with heap-allocations attached to them.

File: arangod/Transaction/Methods.cpp [hunk 2]
```diff
@@ -3976,8 +3972,7 @@ Future<OperationResult> Methods::replaceInternal(
     f = modifyCoordinator(collectionName, newValue, options,
                           TRI_VOC_DOCUMENT_OPERATION_REPLACE, api);
   } else {
-    OperationOptions optionsCopy = options;
-    f = modifyLocal(collectionName, newValue, optionsCopy,
+    f = modifyLocal(collectionName, newValue, options,
```
Comment (jsteemann) [inline arangod/Transaction/Methods.cpp:3975]: nit: is it possible to std::move the `options` here into `modifyLocal(...)`? That would safe a copy of the options object, which contains strings with heap-allocations attached to them.

File: arangod/Transaction/Methods.cpp [hunk 3]
```diff
@@ -3944,8 +3941,7 @@ Future<OperationResult> Methods::updateInternal(
     f = modifyCoordinator(collectionName, newValue, options,
                           TRI_VOC_DOCUMENT_OPERATION_UPDATE, api);
   } else {
-    OperationOptions optionsCopy = options;
-    f = modifyLocal(collectionName, newValue, optionsCopy, /*isUpdate*/ true);
+    f = modifyLocal(collectionName, newValue, options, /*isUpdate*/ true);
```
Comment (jsteemann) [inline arangod/Transaction/Methods.cpp:3944]: nit: is it possible to std::move the `options` here into `modifyLocal(...)`? That would safe a copy of the options object, which contains strings with heap-allocations attached to them.

File: arangod/Transaction/Methods.cpp [hunk 4]
```diff
@@ -2254,7 +2254,7 @@ Future<OperationResult> transaction::Methods::documentCoordinator(
 /// @brief read one or multiple documents in a collection, local
 Future<OperationResult> transaction::Methods::documentLocal(
     std::string const& collectionName, VPackSlice value,
-    OperationOptions const& options) {
+    OperationOptions const options) {
```
[the remark this answers is not in the document]
Comment (goedderz) [reply arangod/Transaction/Methods.cpp:2257]: I generally don't have anything against local variables being const, but OK.

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by goedderz** [comment]
Superseded by https://github.com/arangodb/arangodb/pull/21035

**Pull request closed without merging**